content/kapp-controller/docs/latest/debugging.md

```diff
@@ -0,0 +1,113 @@
+kapp, or helm).
+
+`usefulErrorMessage` will be located at the bottom of the statuses if running a `kubectl get` or `kubectl describe` 
+to view more information about a failure. 
+
+`usefulErrorMessage` can also be accessed more directly through `kubectl` like in the following examples:
+
+```
+# App errors
+$ kubectl get apps/simple-app -o=jsonpath={.status.usefulErrorMessage} -n namespace
+
+# InstalledPackage errors
+$ kubectl get ipkg/simple-app -o=jsonpath={.status.usefulErrorMessage} -n namespace
+
+# PackageRepository errors (cluster scoped so no namespace)
+$ kubectl get pkgr/repo -o=jsonpath={.status.usefulErrorMessage}
+```
+
+### Debugging App CRs
+
+Failures can arise from from fetch, template, deploy, or delete steps for an App CR. These 
+failures correspond to issues with runtime information declared in the App CR's spec. 
+
+Errors are reported as stderr from associated tools used in kapp-controller (i.e. vendir, imgpkg, kbld, ytt, kapp, and helm) 
+or as direct messages from kapp-controller (e.g. when an App uses a ServiceAccount that doesn't exist). 
+
+When a failure occurs with an App CR, you can find further details in the App CR's `DESCRIPTION` column by 
+running `kubectl get apps/simple-app -n namespace`:
+
+```
+NAME         DESCRIPTION                                                                                         SINCE-DEPLOY   AGE
+simple-app   Delete failed: Preparing kapp: Getting service account: serviceaccounts "default-ns-sa" not found   3s             56m
+```
+
+In the case above, the error message shown is coming directly from kapp-controller, so all the information for the failure 
+should be presented in the description column. This commonly occurs when references used by kapp-controller 
+(e.g. secrets, configmaps, serviceaccounts) are not found by kapp-controller.
+
+In cases where the error message does not originate from kapp-controller (e.g. a failed fetch event for a git repository), 
+the stderr from the underlying tool (i.e. vendir, imgpkg, kbld, ytt, kapp, and helm) is shown in the App's status. 
+
+In the App status, there is a field called `usefulErrorMessage` that displays the stderr for a failure during App reconciliation.
+
+This `usefulErrorMessage` field can be found by running `kubectl get apps/simple-app -o=jsonpath={.status.usefulErrorMessage} -n namespace`. 
+The kubectl command will return the stderr output from the App status to help you further understand the reason for the App failure.
+
+The `usefulErrorMessage` can be helpful in pointing out where errors occurred from inputs in the App spec and also 
+pinpoint the resource that caused a deployment failure. However, Apps will not surface errors of resources it is 
+deploying to Kubernetes and further debugging of resources deployed by an App may be needed.
+
+### Debugging InstalledPackage CRs
+
+Failures for InstalledPackages can be directly viewed via the InstalledPackage's status via the `usefulErrorMessage` property of
```

Comment:
> Failures for InstalledPackages can be directly viewed via the InstalledPackage's status via the `usefulErrorMessage` property of the status.

Suggestion: Failures for InstalledPackages can be viewed directly via the `usefulErrorMessage` property of the InstalledPackage's status.